math/fibonacci.c

```diff
@@ -1,23 +1,70 @@
-#include <stdio.h>
+/**
+ * @file
+ * @brief Program to print the nth term of the Fibonacci series.
+ * @details
+ * Fibonacci series generally starts from 0 and 1. Every next term in
+ * the series is equal to the sum of the two preceding terms.
+ */
 
-// Fibonnacci function
-int fib(int number)
+#include <assert.h>  /// for assert()
+#include <stdio.h>   /// for input, output
+#include <stdlib.h>  /// for exit() - to exit the program
+
+/**
+ * @brief Determines the nth Fibonacci term
+ * @param number - n in "nth term" and it can't be negative as well as zero
+ * @return nth term in unsigned type
+ * @warning
+ * Only till 47th and 48th fibonacci element can be stored in 'int' and
+ * 'unsigned' respectively (takes more than 20 seconds to print)
+ */
+unsigned int fib(int number)
 {
-    if (number == 1 || number == 2)
+    // Check for negative integers
+    if (number <= 0)
+    {
+        fprintf(stderr, "Illegal Argument Is Passed!\n");
+        exit(EXIT_FAILURE);
+    }
+
+    // Base conditions
+    if (number == 1)
+        return 0;
+
+    if (number == 2)
         return 1;
-    else
-        return fib(number - 1) + fib(number - 2);
+
+    // Recursive call to the function
+    return fib(number - 1) + fib(number - 2);
+}
+
+/**
+ * @brief self-test implementation
+ * @return void
+ */
+static void test()
+{
+    assert(fib(5));
+    assert(fib(2));
+    assert(fib(9));
```

Comment:
That's not how assertions work, the `assert()` macro is used to test a condition, with the idea being that if the program is working properly, that condition will hold at that point in execution. If you input a function's return value directly, then it doesn't do anything except check if the return value is not `NULL`.
```suggestion
    assert(fib(5) == 3);
    assert(fib(2) == 1);
    assert(fib(9) == 21);
```